nameresolution/local/local.go

```diff
@@ -0,0 +1,85 @@
+// ------------------------------------------------------------
+// Copyright (c) Microsoft Corporation and Dapr Contributors.
+// Licensed under the MIT License.
+// ------------------------------------------------------------
+
+package local
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	ps "github.com/mitchellh/go-ps"
+	"github.com/pkg/errors"
+	process "github.com/shirou/gopsutil/process"
+
+	"github.com/dapr/components-contrib/nameresolution"
+	"github.com/dapr/kit/logger"
+)
+
+type resolver struct {
+	logger logger.Logger
+}
+
+func NewResolver(logger logger.Logger) nameresolution.Resolver {
+	return &resolver{
+		logger: logger,
+	}
+}
+
+func (resolver *resolver) Init(metadata nameresolution.Metadata) error {
+	return nil
+}
+
+func (resolver *resolver) ResolveID(req nameresolution.ResolveRequest) (string, error) {
+	processes, err := ps.Processes()
```

Comment:
Can this be cached? This seem to be a super expensive operation per Invoke() call. Doing it for a cache miss is kind of OK but once cached, we should be able to reuse. Also, invalidating the cache is super important. Maybe we change this logic to be running in the background updating a lookup table and then the ResolveID() method just does a lookup.

Reply:
The value can be cached and we can run a go routine which invalidates the cache every 30 seconds.
We will then need to store process id also in cache, and validate that the process id is actually for dapr process only (it may be that the process id is reassigned)
If the process id does not match then it will start the cache invalidation process again. 
Also, the cache invalidation process will also run only once 30 seconds.

Reply:
Correct. We need to keep track of the PIDs too.